python/hidet/runtime/compiled_task.py

```diff
@@ -213,7 +226,10 @@ def _check_inputs(traced_inputs: Iterable[TensorSignature], inputs):
 
     symbol_map = {}
     for i, (traced, new) in enumerate(zip(traced_inputs, inputs)):
-        if traced.device.partition(':')[0] != new.device.kind:
+        traced_dev_kind = traced.device.partition(':')[0]
+        if traced_dev_kind != new.device.kind:
+            if traced_dev_kind == 'cuda' and new.device.is_vcuda():
+                continue
```

Comment:
Let's add a `target` attribute to the Device class? 

Like 
```python
class Device:
    kind: str  # 'cpu', 'cuda', 'vcuda'
    target: str  # 'cpu', 'cuda'
    id: int  # 0, 1, ...
```
With it, we do not need the special handing here. It will also make the `FlowGraph.build_target` slightly simpler.